code/script.js

```diff
@@ -222,104 +224,172 @@ const generateBoard = () => {
 }
 
-// This function to start (and restart) the game
+// STEP 1 -------This function starts and (and restarts) the game beacuse it is called outside the scope at the bottom of the page start()
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  peopleInPlay = people
+  restartButton.style.display = 'none';
+  generateBoard() //Board is filled with all the cards
+  setSecret() //Secret person is selected
+  selectedQuestion()//functions that stores values from the selected options in the drop down.
 }
+ 
 
-// setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+// STEP 4 ---- the selectedQuestion function gives properties to the currentChioce object
+const selectedQuestion = () => {
 
-  currentQuestion = {
+  const category = choices.options[choices.selectedIndex].parentNode.label//stores the category fron the dropdown
+  console.log(category)
+  const value = choices.options[choices.selectedIndex].value //stores the value from dropdown, for example color of hair if category is hair
+  console.log(value) 
+  currentChoice = { //values are being stored in currentChoice that is a global variable (outside the scope) to be reused further along the code
     category: category,
-    // value: value
+    value: value,
   }
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+// STEP 5 ---- This function should be invoked when you click on 'Find Out' button. Eventlistner is added at the bottom
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentChoice //fetches category and value from currentChioce in the previous step
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
+// Compare the currentChoice attributes with the secret person attributes in a different manner based on category (hair/eyes or accessories/others).
+  // See if we should keep or remove people based on that by sending true or false to filterPeople
   // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
-
+    if (secretPerson[category].includes(value)) {
+      filterPeople(true)
+    } else {
+      filterPeople(false)
+    }
+    console.log("true or false")
   } else if (category === 'accessories' || category === 'other') {
-
+    if (secretPerson[category].includes(value)) {
+  }
+  filterPeople(true); {
```

Comment:
Between the filter people(true) and the filterpeople(false) there are too many curly brackets. Make sure that it looks just as you have done with filterPeople for the values hair and eyes (with the "else" also)